frame/support/procedural/src/construct_runtime/expand/config.rs

```diff
@@ -74,12 +74,18 @@ pub fn expand_outer_config(
 		#[serde(rename_all = "camelCase")]
 		#[serde(deny_unknown_fields)]
 		#[serde(crate = "__genesis_config_serde_import__")]
-		pub struct GenesisConfig {
+		pub struct RuntimeGenesisConfig {
 			#fields
 		}
 
+		// TODO [#14065]
+		// - deprecate it
+		// - later, remove it
```

Comment:
Personally I'd refrain from putting deprecation steps themselves in the code comment and just list them in the issue.

The code comment could say something like "GenesisConfig is in the process of being deprecated [issue link]"